File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/ProductVariant.java
package com.sdl.ecommerce.api.model;

import java.util.List;

/**
 * Product Variant
 *
 * @author nic
 */
public interface ProductVariant {

    /**
     * @return variant product ID
     */
    String getId();

    /**
     * @return variant price
     */
    ProductPrice getPrice();

    /**
     * @return variant attributes
     */
    List<ProductVariantAttribute> getAttributes();
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.Rest/Model/ProductVariant.cs
using SDL.ECommerce.Api.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SDL.ECommerce.Rest.Model
{
    public class ProductVariant : IProductVariant
    {
        public string Id { get; set; }
        public ProductPrice ProductPrice { get; set; }
        public List<ProductAttribute> Attributes { get; set; }

        IProductPrice IProductVariant.Price
        {
            get
            {
                return ProductPrice;
            }
        }

        IList<IProductAttribute> IProductVariant.Attributes
        {
            get
            {
                return Attributes.Cast<IProductAttribute>().ToList();
            }
        }
    }
}
